sbws/core/scanner.py

```diff
@@ -354,6 +354,37 @@ def closure(object):
     return closure
 
 
+def main_loop(args, conf, controller, relay_list, circuit_builder, result_dump,
+              relay_prioritizer, destinations, max_pending_results, pool):
+    pending_results = []
+    while not settings.end_event.is_set():
+        num_relays = 0
+        loop_tstart = time.time()
+        for target in relay_prioritizer.best_priority():
+            if settings.end_event.is_set():
```

Comment:
Why do we need to check settings.end_event.is_set() twice in this function?

Reply:
To don't start a new loop measuring relays (it takes ~3 hours) when sbws is finishing.